dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/passwordreset/PasswordResetPage.java

```diff
@@ -93,6 +99,7 @@ public void init() {
             passwordResetData = passwordResetExecResponse.getPasswordResetData();
             if (passwordResetData != null) {
                 user = passwordResetData.getBuiltinUser();
+                consents = consentService.prepareConsentsForView(session.getLocale());
```

Comment:
terms of use is showing when `isAccountUpgrade()` is true. That is more restrictive case than only `passwordResetData != null` there should be also checking of `passwordResetData.getReason() == PasswordResetData.Reason.UPGRADE_REQUIRED;` without this check there will be easy to make a mistake of saving some accepted consents without really having user accepting any consent.

Reply:
I was thinking about it but I thought that since we are filtering by accepted boolean (false by default) in service, there shouldn't be any consent saved if we are not going to show them. But I guess adding if here would also give a clue when consents are used on this Page.